accounts-db/src/account_locks.rs

```diff
@@ -0,0 +1,162 @@
+#[cfg(feature = "dev-context-only-utils")]
+use qualifier_attr::qualifiers;
+use {
+    ahash::{AHashMap, AHashSet},
+    log::debug,
+    solana_sdk::{pubkey::Pubkey, transaction::TransactionError},
+    std::collections::hash_map,
+};
+
+#[derive(Debug, Default)]
+pub struct AccountLocks {
+    write_locks: AHashSet<Pubkey>,
+    readonly_locks: AHashMap<Pubkey, u64>,
+}
+
+impl AccountLocks {
+    /// Lock the account keys in `keys` for a transaction.
+    /// The bool in the tuple indicates if the account is writable.
+    /// Returns an error if any of the accounts are already locked in a way
+    /// that conflicts with the requested lock.
+    pub fn try_lock_accounts_for_transaction<'a>(
+        &mut self,
+        keys: impl Iterator<Item = (&'a Pubkey, bool)> + Clone,
+    ) -> Result<(), TransactionError> {
+        for (key, writable) in keys.clone() {
+            if writable {
+                if !self.can_write_lock(key) {
+                    debug!("Writable account in use: {:?}", key);
```

Comment:
nit: maybe remove this debug line, doubt it is monitored at all.

Reply:
16d492d9e0b260403d853c9e5fe972c82648ef71